modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -165,6 +171,64 @@ export default class MVTLayer extends TileLayer {
       feature => getFeatureUniqueId(feature, uniqueIdProperty) === featureIdToHighlight
     );
   }
+
+  _pickObjects(maxFeatures) {
+    const {deck, viewport} = this.context;
+    const width = viewport.width;
+    const height = viewport.height;
+    const x = viewport.x;
+    const y = viewport.y;
+    const layerIds = [this.id];
+    return deck.pickObjects({x, y, width, height, layerIds, maxFeatures});
+  }
+
+  getRenderedFeatures({format = 'json', maxFeatures = null}) {
+    const features = this._pickObjects(maxFeatures);
+    const featureCache = new Set();
+    const renderedFeatures = [];
+
+    for (const f of features) {
+      const featureId = getFeatureUniqueId(f.object, this.props.uniqueIdProperty);
+
+      if (featureId === -1) {
+        // we have no id for the feature, we just add to the list
+        renderedFeatures.push(getFeatureByFormat(f.object, format));
+      } else if (!featureCache.has(featureId) && featureId !== -1) {
+        // Add removing duplicates
+        featureCache.add(featureId);
+        renderedFeatures.push(getFeatureByFormat(f.object, format));
```

Comment:
What is the use case for the `format` argument? It's no longer "features" that are returned if you take out just the properties. There's also no perf savings since all features are already stored in memory and will be passed by reference.

Reply:
After a second thought, I've just removed it to make the interface simpler. We always return the geometry + attributes and the users can filter out based on their use case.